app/src/main/java/com/portfolio/tasky/di/modules/AppModule.kt

```diff
@@ -0,0 +1,28 @@
+package com.portfolio.tasky.di.modules
+
+import android.app.Application
+import android.content.Context
+import androidx.lifecycle.LiveData
+import com.portfolio.tasky.usecases.NetworkChangeReceiver
+import com.portfolio.tasky.usecases.NetworkStatus
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import javax.inject.Singleton
+
+@Module
+@InstallIn(SingletonComponent::class)
+class AppModule {
+
+    @Provides
+    fun providesContext(application: Application): Context {
+        return application.applicationContext
+    }
+
+    @Singleton
+    @Provides
+    fun provideNetworkChangeReceiver(context: Context): LiveData<NetworkStatus> {
```

Comment:
Here you could simply add the ApplicationContext annotation, and it saves you to provide the context manually like you did before

Like this:
```
fun provideNetworkChangeReceiver(@ApplicationContext context: Context)
```

Reply:
Interesting. Sounds good then, let's leave it as it is